mobile/android/android-emulator-services.ts

```diff
@@ -45,6 +45,21 @@ class AndroidEmulatorServices implements Mobile.IEmulatorPlatformServices {
 		this.adbFilePath = this.$staticConfig.getAdbFilePath().wait();
 	}
 
+	public getEmulatorId(): IFuture<string> {
+		return (() => {
+			let image = this.getEmulatorImage().wait();
+			let emulatorId: string = null;
+
+			if(image) {
```

Comment:
Not a merge stopper, but this code could be simplified by checking for the error condition first:

```
if (!image) fail();
return this.startInstance().wait();
```